src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/Interop/SNINativeMethodWrapper.cs

```diff
@@ -397,9 +397,6 @@ internal struct SNI_Error
         #endregion
 
         #region DLL Imports
-        [DllImport("kernel32.dll", SetLastError = true)]
-        public static extern IntPtr LoadLibrary(string dllToLoad);
-
         [DllImport("secur32.dll", ExactSpelling = true, SetLastError = true)]
         internal static extern uint QueryContextAttributes(ref CredHandle contextHandle, [In] ContextAttribute attribute, [In] IntPtr buffer);
```

Comment:
This method, along with all its references, can be deleted as it is a leftover from the old CertAuthentication implementation, which is no longer in use. The entire section in TDSParser that validates the server certificate can also be removed.
```C#
// Validate server certificate
if (serverCallback != null)
{
    X509Certificate2 serverCert = null;

    error = SNINativeMethodWrapper.SNISecGetServerCertificate(_physicalStateObj.Handle, ref serverCert);
    if (error != TdsEnums.SNI_SUCCESS)
    {
        _physicalStateObj.AddError(ProcessSNIError(_physicalStateObj));
        ThrowExceptionAndWarning(_physicalStateObj);
    }

    bool valid = serverCallback(serverCert);
    if (!valid)
    {
        throw SQL.InvalidServerCertificate();
    }
}
```

Reply:
Are there definitely no internal builds with ADONET_CERT_AUTH enabled? If there aren't, I'll remove that implementation in a separate PR so that this one's easier to review.

Does the same apply to ADONET_ORIGINAL_CLIENT_ADDRESS and CTAIP support?

Reply:
> Are there definitely no internal builds with ADONET_CERT_AUTH enabled? If there aren't, I'll remove that implementation in a separate PR so that this one's easier to review.
> 
> Does the same apply to ADONET_ORIGINAL_CLIENT_ADDRESS and CTAIP support?

All ADONET_CERT_AUTH related items can be safely removed. Thanks, that would be a big help—this has been on our to-do list for a long time.

Reply:
> Does the same apply to ADONET_ORIGINAL_CLIENT_ADDRESS and CTAIP support?

Correct. Those could be also removed.

Reply:
I've submitted that as #2831.